sdks/java/io/cassandra/build.gradle

```diff
@@ -17,7 +17,24 @@
  */
 
 apply plugin: org.apache.beam.gradle.BeamModulePlugin
-applyJavaNature()
+applyJavaNature(shadowClosure: {
+    dependencies {
+        include(dependency(project.library.java.guava))
+    }
+    // guava uses the com.google.common and com.google.thirdparty package namespaces
+    relocate("com.google.common", project.getJavaRelocatedPath("com.google.common")) {
+        // com.google.common is too generic, need to exclude guava-testlib
+        exclude "com.google.common.collect.testing.**"
+        exclude "com.google.common.escape.testing.**"
+        exclude "com.google.common.testing.**"
+        exclude "com.google.common.util.concurrent.testing.**"
+        // don't relocate because the cassandra driver's public API uses it
```

Comment:
I think this is safe, but weird. This will result in a `beam-io-cassandra` jar that contains relocated class files for most of Guava, plus a `ListenableFuture.class` at the normal package path but with all internal deps relocated to point to the rest of the bundled Guava classes.

If another totally unmodified Guava takes precedence, then _that_ ListenableFuture will be used, and it will point at the unmodified other Guava classes. I think the result is indistinguishable, but it took me a bit to think that through so I might have got the details wrong.

Overall, since the user is going to have Guava on their classpath, I would just not relocate it at all. Any uses of Guava that have nothing to do with Cassandra can be ported to `beam-vendored-guava-20` so they are even more obviously disjoint and unrelated.

Reply:
If I understand this correctly, the solution is to:
1. remove the guava relocations in `beam-io-cassandra`
2. add `beam-vendored-guava-20` as a `shadow` dependency and move any references *within `beam-io-cassandra`* to that

Reply:
Yes, exactly